front/lib/actions/mcp_internal_actions/input_configuration.ts

```diff
@@ -192,49 +146,66 @@ function generateConfiguredInput({
 
     case INTERNAL_MIME_TYPES.TOOL_INPUT.STRING: {
       // For primitive types, we have rendered the key from the path and use it to look up the value.
-      const value = getValidatedValue<string>(
-        actionConfiguration,
-        keyPath,
-        actionConfiguration.additionalConfiguration[keyPath],
-        "string",
-        (val): val is string => typeof val === "string"
-      );
+      let value: JSONSchema7Type | AdditionalConfigurationValueType | null =
+        actionConfiguration.additionalConfiguration[keyPath];
 
+      if (value === undefined) {
+        value = getDefaultValue(actionConfiguration, keyPath);
+      }
+
+      if (typeof value !== "string") {
+        throw new Error(
+          `Expected string value for key ${keyPath}, got ${typeof value}`
+        );
+      }
```

Comment:
nit: would `assert` instead of `throw` here given that we `assert` elsewhere in the file for the same kind of thing
```suggestion
      assert(isString(value), `Expected string value for key ${keyPath}, got ${typeof value}`);
```